scapy/arch/pcapdnet.py

```diff
@@ -455,8 +455,7 @@ def next(self):
                 __next__ = next
 
                 def fileno(self):
-                    raise RuntimeError("%s has no fileno. Please report this bug." %
-                                       self.__class__.__name__)
+                    return self.pcap.getfd()
```

Comment:
I think it would be a good idea to use something similar to
https://github.com/secdev/scapy/blob/master/scapy/arch/pcapdnet.py#L464-L469

Reply:
Done !